sql/analyzer/indexed_joins.go

```diff
@@ -597,8 +597,12 @@ func getJoinIndexes(
 	case *expression.And:
 		exprs := splitConjunction(joinCond.cond)
 		for _, expr := range exprs {
-			switch expr.(type) {
-			case *expression.Equals, *expression.NullSafeEquals:
+			switch e := expr.(type) {
+			case *expression.Equals, *expression.NullSafeEquals, *expression.IsNull:
+			case *expression.Not:
```

Comment:
This works because `getJoinIndex` below is going to look for a complete index and then it's going to take the first partial index that matches. A partial index and applying the conjunction filter after the fact is the better path here.

It's not clear to me why we are so strict in the types of expressions we go after here...if there was an index that matched the expression `a.blah LIKE b.like` or whatever, wouldn't we want to use it just the same? Isn't any partial index on any portion of the conjunction still going to be correct and be better than no index at all?

Reply:
We could definitely loosen this up significantly, no reason to restrict it to only equality